src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/SemanticTokensRefreshEndpoint.cs

```diff
@@ -23,12 +24,21 @@ public SemanticTokensRefreshEndpoint(WorkspaceSemanticTokensRefreshPublisher sem
             _semanticTokensRefreshPublisher = semanticTokensRefreshPublisher;
         }
 
-        public Task<Unit> Handle(SemanticTokensRefreshParams request, CancellationToken cancellationToken)
+        public Task<Unit> Handle(SemanticTokensRefreshParamsBridge request, CancellationToken cancellationToken)
         {
             // We have to invalidate the tokens cache since it may no longer be up to date.
             _semanticTokensRefreshPublisher.EnqueueWorkspaceSemanticTokensRefresh();
 
             return Unit.Task;
         }
+
+        public RegistrationExtensionResult? GetRegistration(VSInternalClientCapabilities clientCapabilities)
+        {
+            const string ServerCapability = "workspace.semanticTokens";
+
+            return new RegistrationExtensionResult(ServerCapability, new SemanticTokenRefreshRegistrationOptions(RefreshSupport: true));
+        }
+
+        private record SemanticTokenRefreshRegistrationOptions(bool RefreshSupport);
```

Comment:
Ah is there no VS equivalent?

Reply:
As above, it exists, I just didn't search right lol.

Reply:
Ya their naming is a bit odd. One thing I've gotten accustomed to is searching for the LSP string representation of the sub-properties since there's some part of code that needs to state that 😄